src/components/macro/macro-action-editor/tab/macro-key/macro-key.component.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    Component,
+    OnInit,
+    Input,
+    ViewChild
+} from '@angular/core';
+
+import {Tab} from '../../../../popover/tab/tab';
+import {MacroAction, macroActionType} from '../../../../../config-serializer/config-items/MacroAction';
+
+import { PressKeyMacroAction } from '../../../../../config-serializer/config-items/PressKeyMacroAction';
+import { HoldKeyMacroAction } from '../../../../../config-serializer/config-items/HoldKeyMacroAction';
+import { ReleaseKeyMacroAction } from '../../../../../config-serializer/config-items/ReleaseKeyMacroAction';
+
+import {KeypressTabComponent} from '../../../../popover/tab/keypress/keypress-tab.component';
+
+enum TabName {
+    Keypress,
+    Hold,
+    Release
+}
+
+@Component({
+    selector: 'macro-key-tab',
+    template: require('./macro-key.component.html'),
+    styles: [
+        require('../../macro-action-editor.component.scss'),
+        require('./macro-key.component.scss')
+    ],
+    host: { 'class': 'macro__mouse' }
+})
+export class MacroKeyTabComponent implements OnInit {
+    @Input() macroAction: MacroAction;
+    @ViewChild('tab') selectedTab: Tab;
+
+    private activeTab: TabName;
+    private tabName = TabName;
```

Comment:
We should be consistent. Enums should start with uppercase, however using enums directly in the template is not possible, since we can only use members, so we should hack it this way. I think the comment gives explanation for those, who see the code at the first time.